eng/targets/Settings.props

```diff
@@ -194,4 +194,10 @@
       </PropertyGroup>
     </When>
   </Choose>
+
+  <PropertyGroup Condition="'$(DotNetBuildFromSource)' == 'true'">
+    <!-- https://github.com/dotnet/roslyn/issues/38433 Vbc does not like the extra semicolon -->
+    <DefineConstants Condition="'$(DefineConstants)' != ''">DOTNET_BUILD_FROM_SOURCE;$(DefineConstants)</DefineConstants>
+    <DefineConstants Condition="'$(DefineConstants)' == ''">DOTNET_BUILD_FROM_SOURCE</DefineConstants>
```

Comment:
@tmat is there a reason that arcade doesn't already provide this? Seems like a small service we should centralize as I imagine more and more repositories will end up needing to do this.

Reply:
This shouldn't be used much. In general I don't think we want source build to produce assemblies that have different functionality from regular build. 
I haven't seen any other project that would need it.